wordpress-mobile/WordPress-Android
# Setup UI layer for new MySite screen

Fixes #13341 and #13342 

I've used this PR to handle 2 issues (cause they seemed really related). I've created an Adapter, a DiffCallback, ViewModel, put everything into the Fragment so it should be ready for implementation of all the basic list items. 

To test:
- Turn on the MySiteImprovements flag (App settings/Test feature configuration)
- Restart the app
- Notice the empty My Site fragment
- Notice nothing is crashing

PR submission checklist:
